File: 프로그래머스/1/42748. K번째수/K번째수.js
function solution(array, commands) {
    
    //[i,j,k] i번째에서 j번째까지 array 자르고, 자른 배열에서 k번째 수 가져오기
    
    let result = [];
    
    for(let a = 0; a < commands.length; a++){
        let i = commands[a][0]; 
        let j = commands[a][1];
        let k = commands[a][2];
        
        if(i == j){
            result.push(array[i-1]);
        }else{
            let temp = array.slice(i-1, j);
            temp.sort((a,b) => a-b);
            result.push(temp[k-1]);
        }
    }
    
    return result;
}

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        
        int[] answer = new int[commands.length]; //정답 배열
        
        
        for(int i = 0; i < commands.length; i++){
            int[] temp = commands[i];
            
            answer[i] = sliceArray(array, temp);
        }
        
        return answer;
    }
    
    public static int sliceArray(int[] array, int[] temp){
        
        int i = temp[0]-1;
        int j = temp[1]-1;
        int k = temp[2]-1;
        
        int[] slice = new int[j-i+1];
        
        for(int index = 0; index < slice.length; index++){
            
            slice[index] = array[i++];
            
            if(j < i){
                break;
            }
        }
        
        Arrays.sort(slice);
        
        return slice[k];
    }
    
}